app/controllers/api/service_templates_controller.rb

```diff
@@ -4,5 +4,57 @@ class ServiceTemplatesController < BaseController
     include Subcollections::Tags
     include Subcollections::ResourceActions
     include Subcollections::ServiceRequests
+
+    def create_resource(_type, _id, data)
+      # Temporarily only supporting atomic.
+      raise 'Service Type composite not supported' if data['service_type'] == 'composite'
+      create_atomic(data).tap(&:save!)
+    rescue => err
+      raise BadRequestError, "Could not create Service Template - #{err}"
+    end
+
+    private
+
+    def create_atomic(data)
+      service_template = ServiceTemplate.new(data.except('config_info'))
+      config_info = data['config_info'].except('ae_endpoints') || {}
+      case service_template.type
+      when 'ServiceTemplateOrchestration'
```

Comment:
@jntullo We can merge this one if it becomes a blocker for other development work; otherwise I would think this is the right place to refactor.

I started to add the following to `service_template.rb`
```
  # create ServiceTemplate and supporting ServiceResources and ResourceActions
  def self.create_catalog_item(options)

  end
```
The implementation is empty right now. You can move logic to this block and corresponding subclasses.
The method should return an instance of `ServiceTemplate` or subclass.

The method name can be debatable; I could not think of a better name.